Dockerfile

```diff
@@ -1,78 +1,28 @@
-#**********************************************************************
-# Builder
-# Create a go runtime suitable for building and testing kfctl
-ARG GOLANG_VERSION=1.13.7
-FROM golang:$GOLANG_VERSION as builder
-
-ARG BRANCH=master
-ARG REPO=https://github.com/kubeflow/kubeflow
-
-RUN apt-get update
-RUN apt-get install -y git unzip jq vim
-
-# junit report is used to conver go test output to junit for reporting
-RUN go get -u github.com/jstemmer/go-junit-report
-
-# We need gcloud to get gke credentials.
-RUN if [ "$(uname -m)" = "x86_64" ]; then \
-        cd /tmp && \
-        wget -nv https://dl.google.com/dl/cloudsdk/release/install_google_cloud_sdk.bash && \
-        chmod +x install_google_cloud_sdk.bash && \
-        ./install_google_cloud_sdk.bash --disable-prompts --install-dir=/opt/; \
-    fi
-
-ENV PATH /go/bin:/usr/local/go/bin:/opt/google-cloud-sdk/bin:${PATH}
-
-# use go modules
-ENV GO111MODULE=on
-ENV GOPATH=/go
-# Workaround for https://github.com/kubernetes/gengo/issues/146
-ENV GOROOT=/usr/local/go
-
-# Create kfctl folder
-RUN mkdir -p ${GOPATH}/src/github.com/kubeflow/kfctl
-WORKDIR ${GOPATH}/src/github.com/kubeflow
-RUN mkdir kubeflow
-RUN echo REPO=${REPO} branch=${BRANCH}
-RUN git clone ${REPO} --depth=1 --branch ${BRANCH} --single-branch kubeflow
-WORKDIR ${GOPATH}/src/github.com/kubeflow/kfctl
-
-# Download dependencies first to optimize Docker caching.
-COPY go.mod .
-COPY go.sum .
+# Build the manager binary
+ARG GOLANG_VERSION=1.18.4
+FROM registry.access.redhat.com/ubi8/go-toolset:$GOLANG_VERSION as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
 RUN go mod download
-# Copy in the source
-COPY . .
-
-#**********************************************************************
-#
-# kfctl_base
-#
-FROM builder as kfctl_base
-
-RUN make build-kfctl && \
-    if [ "$(uname -m)" = "aarch64" ]; then \
-        cp bin/arm64/kfctl bin/kfctl; \
-    fi
-
-#**********************************************************************
-#
-# Final image base
-#
 
-FROM alpine:3.10.1 as barebones_base
-RUN mkdir -p /opt/kubeflow
-WORKDIR /opt/kubeflow
+# Copy the go source
+COPY main.go main.go
+COPY apis/ apis/
+COPY controllers/ controllers/
+COPY pkg/ pkg/
 
-#**********************************************************************
-#
-# kfctl
-#
-FROM barebones_base as kfctl
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
```

Comment:
With the change to ubi8/go-toolset, this line is throwing an error
```
[1/2] STEP 10/10: RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
controllers/kfdef.apps.kubeflow.org/kfdef_controller.go:61:2: open /opt/odh-operator/pkg/kfapp/coordinator: permission denied
controllers/kfdef.apps.kubeflow.org/kfdef_controller.go:62:2: open /opt/odh-operator/pkg/kfconfig/loaders: permission denied
controllers/kfdef.apps.kubeflow.org/kfdef_controller.go:63:2: open /opt/odh-operator/pkg/utils: permission denied
Error: building at STEP "RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go": while running runtime: exit status 1
```

Reply:
This is is due to the builder image running as the `default` user but the 'WORKDIR` `/workspace` has `755` permissions preventing the `go` command from creating files.